[DOCS-1540] Initial docs for LLM Evaluation Jobs (#1910)
wandb/docs: models/launch/evaluate-hosted-model.mdx

## Prerequisites
1. Review the [requirements and limitations](/models/launch#more-details) for LLM Evaluation Jobs.
1. A team admin must add an OpenAI API key as a [team-scoped secret](/platform/secrets#add-a-secret). This is required by many benchmarks for scoring. Any team member can specify the secret when configuring an evaluation job.
1. To authenticate to the hosted API model endpoint, a team admin must add the API key as a [team-scoped secret](/platform/secrets#add-a-secret).
1. Review the documentation for a given benchmark to understand how it works and learn about specific requirements. For convenience, the [Available evaluation benchmarks](/models/launch/evaluations) reference includes relevant links.

## Evaluate your model
Follow these steps to set up and launch an evaluation:

1. Log in to W&B, then click **Launch** in the left navigation. The **LLM Evaluation Jobs** page displays.
1. Click **Evaluate hosted API model** to set up the evaluation.
1. Select a destination project where the evaluation run and results will be saved.
1. In the **Model** section, specify the base URL and model name to evaluate, and select the API key to use for authentication.

> 1. Click **Select evaluations**, then select up to six benchmarks to run.

Reviewer: The maximum is 4 benchmarks

---
description: "Evaluate a hosted API model using infrastructure managed by CoreWeave"
title: "Evaluate a hosted API model"
---
import ReviewEvaluationResults from "/snippets/en/_includes/llm-eval-jobs/review-evaluation-results.mdx";
import RerunEvaluation from "/snippets/en/_includes/llm-eval-jobs/rerun-evaluation.mdx";
import ExportEvaluation from "/snippets/en/_includes/llm-eval-jobs/export-evaluation.mdx";
import PreviewLink from '/snippets/en/_includes/llm-eval-jobs/preview.mdx';

<PreviewLink />

This page shows how to use [W&B LLM Evaluation Jobs](/models/launch) to run a series of evaluation benchmarks on a hosted API model at a publicly-accessible URL, using infrastructure managed by CoreWeave. To evaluate a model checkpoint instead, see [Evaluate a model checkpoint](/models/launch/evaluate-model-checkpoint) instead.

## Prerequisites
1. Review the [requirements and limitations](/models/launch#more-details) for LLM Evaluation Jobs.
1. To run certain benchmarks, a team admin must add the required API keys as [team-scoped secrets](/platform/secrets#add-a-secret). Any team member can specify the secret when configuring an evaluation job.

>     - **OpenAPI API key**: Used by many benchmarks for scoring. Required if the field **Scorer API key** appears after you select a benchmark.

Reviewer 2: nit: "An OpenAI Api Key"? or `OPENAI_API_KEY`

Reviewer 2: Generally all secret names must match the expected name for the api key for the model provider.

OpenAI -> OPENAI_API_KEY
Anthropic -> ANTHROPIC_API_KEY

Reviewer 2: So the scorer must be called OPENAI_API_KEY

      - To use the OpenRouter provider, prefix the model name with `openrouter` in the format `openrouter/<model-name>`.
      - To evaluate a custom OpenAPI-compliant model, specify the model name in the format `openai-api/<model-name>`.
1. Click **Select evaluations**, then select up to four benchmarks to run.
1. If you select one or more benchmarks that require an OpenAI API key for scoring, the **Scorer API key** field displays. Click it, then select the OpenAI API key's secret by name. For convenience, a team admin can create a secret from this drawer by clicking **Create secret**.
1. Optionally, set **Sample limit** to a positive integer to limit the maximum number of benchmark samples to evaluate. Otherwise, all samples in the task are included.
1. To create a leaderboard automatically, click **Publish results to leaderboard**. The leaderboard will display all evaluations together in a workspace panel, and you can also share it in a report.
1. Click **Launch** to launch the evaluation job.

> 1. Click the circular arrow icon at the top of the page to open the recent run modal. Evaluation jobs appear with other runs in the project. Click the name of a finished run to open it in single-run view, or click the **Leaderboard** link to open the leaderboard directly. For details, see [View the results](#view-the-results).

Reviewer 2: "with other runs in the project"
The modal isn't restricted to a single project, it should show all of a users recent runs

---
description: "Evaluate a hosted API model using infrastructure managed by CoreWeave"
title: "Evaluate a hosted API model"
---
import ReviewEvaluationResults from "/snippets/en/_includes/llm-eval-jobs/review-evaluation-results.mdx";
import RerunEvaluation from "/snippets/en/_includes/llm-eval-jobs/rerun-evaluation.mdx";
import ExportEvaluation from "/snippets/en/_includes/llm-eval-jobs/export-evaluation.mdx";
import PreviewLink from '/snippets/en/_includes/llm-eval-jobs/preview.mdx';

> <PreviewLink />

Author: ```suggestion
<Note><PreviewLink /></Note>
```

> This page shows how to use [W&B LLM Evaluation Jobs](/models/launch) to run a series of evaluation benchmarks on a hosted API model at a publicly-accessible URL, using infrastructure managed by CoreWeave. To evaluate a model checkpoint instead, see [Evaluate a model checkpoint](/models/launch/evaluate-model-checkpoint) instead.

Reviewer 3: ```suggestion
This page shows how to use [W&B LLM Evaluation Jobs](/models/launch) to run a series of evaluation benchmarks on a hosted API model at a publicly-accessible URL, using infrastructure managed by CoreWeave. To evaluate a model checkpoint saved as a W&B artifact, see [Evaluate a model checkpoint](/models/launch/evaluate-model-checkpoint) instead.
```

1. The model to evaluate must be available at a publicly accessible URL. An organization or team admin must create a team-scoped secret with the API key for authentication.
1. Create a new [W&B project](/models/track/project-page) for the evaluation results. From the left navigation, click **Create new project**.
1. Review the documentation for a given benchmark to understand how it works and learn about specific requirements. For convenience, the [Available evaluation benchmarks](/models/launch/evaluations) reference includes relevant links.

## Evaluate your model
Follow these steps to set up and launch an evaluation job:

1. Log in to W&B, then click **Launch** in the left navigation. The **LLM Evaluation Jobs** page displays.
1. Click **Evaluate hosted API model** to set up the evaluation.

> 1. Select a destination project where the evaluation results will be saved.

Reviewer 3: ```suggestion
1. Select a destination project to save the evaluation results to.
```
nit and totally optional

1. In the **Model** section, specify the base URL and model name to evaluate, and select the API key to use for authentication.

>       - To evaluate a model provided by [W&B Inference](/inference), set the base URL to `https://api.inference.wandb.ai/v1` and specify the model name in the format `openai-api/wandb/<model_id>`. Refer to the [Inference model catalog](/inference/models) for details.

Reviewer 3: ```suggestion
      - To evaluate a model provided by [W&B Inference](/inference), set the base URL to `https://api.inference.wandb.ai/v1` and specify the model name in the following syntax: `openai-api/wandb/<model_id>`. Refer to the [Inference model catalog](/inference/models) for details.
```

To me, this makes more sense, but it's ultimately a nit. You'd have to update the bullets below if you wanted to use it, too. Up to you

>       - To evaluate a model provided by [W&B Inference](/inference), set the base URL to `https://api.inference.wandb.ai/v1` and specify the model name in the following syntax: `openai-api/wandb/<model_id>`. Refer to the [Inference model catalog](/inference/models) for details.

Reviewer 2: Thinking about this more, we should link to the inspect AI evals model providers docs so users can name their models. https://inspect.aisi.org.uk/providers.html#openai-api

that way it explains how it works for wandb inference, and any other non-standard model providers (also tells them how to use each of the model providers that are supported by default

>       - To evaluate a publicly hosted model, prefix the model name with the model provider. For example, specify an OpenAI model using the format `openai/<model-name>`. For a comprehensive list of hosted model providers and models, see [AI Security Institute's model provider reference](https://inspect.aisi.org.uk/providers.html).

Author: ```suggestion
      - To evaluate a publicly hosted model, prefix the model name with the model provider. For example, specify an OpenAI mode in the following syntax:  `openai/<model-name>`. For a comprehensive list of hosted model providers and models, see [AI Security Institute's model provider reference](https://inspect.aisi.org.uk/providers.html).
```

>       - To use the OpenRouter provider, prefix the model name with `openrouter` in the format `openrouter/<model-name>`.

Author: ```suggestion
      - To use the OpenRouter provider, prefix the model name with `openrouter` in the following syntax: `openrouter/<model-name>`.
```

>       - To evaluate a custom OpenAPI-compliant model, specify the model name in the format `openai-api/<model-name>`.

Author: ```suggestion
      - To evaluate a custom OpenAPI-compliant model, specify the model name in the following syntax: `openai-api/<model-name>`.
```

1. Click **Select evaluations**, then select up to four benchmarks to run.

> 1. If you select benchmarks that use OpenAI models for scoring,, the **Scorer API key** field displays. Click it, then select the `OPENAI_API_KEY` secret. For convenience, a team admin can create a secret from this drawer by clicking **Create secret**.

Reviewer 3: ```suggestion
1. If you select benchmarks that use OpenAI models for scoring, the **Scorer API key** field displays. Click it, then select the `OPENAI_API_KEY` secret. For convenience, a team admin can create a secret from this drawer by clicking **Create secret**.
```
